ci-operator/step-registry/upi/install/azure/upi-install-azure-commands.sh

```diff
@@ -257,8 +255,9 @@ echo "Deploying 04_bootstrap"
 BOOTSTRAP_URL=$(az storage blob url --account-name $ACCOUNT_NAME --account-key $ACCOUNT_KEY -c "files" -n "bootstrap.ign" -o tsv)
 IGNITION_VERSION=$(jq -r .ignition.version ${ARTIFACT_DIR}/installer/bootstrap.ign)
 BOOTSTRAP_IGNITION=$(jq -rcnM --arg v "${IGNITION_VERSION}" --arg url $BOOTSTRAP_URL '{ignition:{version:$v,config:{replace:{source:$url}}}}' | base64 -w0)
+# shellcheck disable=SC2046
 az deployment group create -g $RESOURCE_GROUP \
-  --template-file "04_bootstrap.json" \
+  --template-file "04_bootstrap.json" $([ ${#BOOTSTRAP_INSTANCE_TYPE} -gt 0 ] && echo "--parameters bootstrapVMSize=${BOOTSTRAP_INSTANCE_TYPE}") \
```

Comment:
Another option:
```
az_deployment_bootstrap_instance=""
if [ -n "${BOOTSTRAP_INSTANCE_TYPE}" ]; then
    az_deployment_bootstrap_instance="--parameter bootstrapVMSize=${BOOTSTRAP_INSTANCE_TYPE}"
fi
[...]
  --parameters baseName="$INFRA_ID" ${az_deployment_bootstrap_instance} ${az_deployment_optional_parameters}
```

Reply:
I would have avoided the command substitution as well as the definition of a new var. Let me see and propose something else.

Reply:
As disscussed offline, changed the `len > 0` condition to `-n "$...`